contracts/settling_game/L03_Buildings.cairo

```diff
@@ -429,57 +299,120 @@ func get_buildings_unpacked{
 }(token_id : Uint256) -> (realm_buildings : RealmBuildings):
     alloc_locals
 
-    let (data) = get_realm_buildings(token_id)
-
-    let (Fairgrounds) = unpack_data(data, 0, 63)
-    let (RoyalReserve) = unpack_data(data, 6, 63)
-    let (GrandMarket) = unpack_data(data, 12, 63)
-    let (Castle) = unpack_data(data, 18, 63)
-    let (Guild) = unpack_data(data, 24, 63)
-    let (OfficerAcademy) = unpack_data(data, 30, 63)
-    let (Granary) = unpack_data(data, 36, 63)
-    let (Housing) = unpack_data(data, 42, 63)
-    let (Amphitheater) = unpack_data(data, 48, 63)
-    let (ArcherTower) = unpack_data(data, 54, 63)
-    let (School) = unpack_data(data, 60, 63)
-    let (MageTower) = unpack_data(data, 66, 63)
-    let (TradeOffice) = unpack_data(data, 72, 63)
-    let (Architect) = unpack_data(data, 78, 63)
-    let (ParadeGrounds) = unpack_data(data, 84, 63)
-    let (Barracks) = unpack_data(data, 90, 63)
-    let (Dock) = unpack_data(data, 96, 63)
-    let (Fishmonger) = unpack_data(data, 102, 63)
-    let (Farms) = unpack_data(data, 108, 63)
-    let (Hamlet) = unpack_data(data, 114, 63)
+    let (data) = get_storage_realm_buildings(token_id)
+
+    let (House) = unpack_data(data, 0, 63)
+    let (StoreHouse) = unpack_data(data, 6, 63)
+    let (Granary) = unpack_data(data, 12, 63)
+    let (Farm) = unpack_data(data, 18, 63)
+    let (FishingVillage) = unpack_data(data, 24, 63)
+    let (Barracks) = unpack_data(data, 30, 63)
+    let (MageTower) = unpack_data(data, 36, 63)
+    let (ArcherTower) = unpack_data(data, 42, 63)
+    let (Castle) = unpack_data(data, 48, 63)
 
     return (
         realm_buildings=RealmBuildings(
-        Fairgrounds=Fairgrounds,
-        RoyalReserve=RoyalReserve,
-        GrandMarket=GrandMarket,
-        Castle=Castle,
-        Guild=Guild,
-        OfficerAcademy=OfficerAcademy,
+        House=House,
+        StoreHouse=StoreHouse,
         Granary=Granary,
-        Housing=Housing,
-        Amphitheater=Amphitheater,
+        Farm=Farm,
+        FishingVillage=FishingVillage,
+        Barracks=Barracks,
+        MageTower=MageTower,
         ArcherTower=ArcherTower,
-        School=School,
+        Castle=Castle
+        ),
+    )
+end
+
+@view
+func get_buildings_time{
+    syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr, bitwise_ptr : BitwiseBuiltin*
+}(token_id : Uint256) -> (realm_buildings : RealmBuildings):
+    alloc_locals
+
+    # TODO: Hardcoded only castle felt for testing, pack buildings into typeof felt
+
+    let (House) = castle_time.read(token_id)
+    let (StoreHouse) = castle_time.read(token_id)
+    let (Granary) = castle_time.read(token_id)
+    let (Farm) = castle_time.read(token_id)
+    let (FishingVillage) = castle_time.read(token_id)
+    let (Barracks) = castle_time.read(token_id)
+    let (MageTower) = castle_time.read(token_id)
+    let (ArcherTower) = castle_time.read(token_id)
+    let (Castle) = castle_time.read(token_id)
```

Comment:
What's the reason we can't pack these into one felt?

Reply:
timestamp takes 25 bits, so can only fit 6 in a felt. The plan was to move these into 2 felts:
- Combat
- Economy + Housing

Just leaving this to last